exchanges/stream/websocket_connection.go

```diff
@@ -90,25 +90,22 @@ func (w *WebsocketConnection) Dial(dialer *websocket.Dialer, headers http.Header
 // SendJSONMessage sends a JSON encoded message over the connection
 func (w *WebsocketConnection) SendJSONMessage(data interface{}) error {
 	if !w.IsConnected() {
-		return fmt.Errorf("%s websocket connection: cannot send message to a disconnected websocket",
-			w.ExchangeName)
+		return fmt.Errorf("%s websocket connection: cannot send message to a disconnected websocket", w.ExchangeName)
 	}
 
 	w.writeControl.Lock()
 	defer w.writeControl.Unlock()
 
 	if w.Verbose {
-		log.Debugf(log.WebsocketMgr,
-			"%s websocket connection: sending message to websocket %+v\n",
-			w.ExchangeName,
-			data)
+		if msg, err := json.Marshal(data); err == nil { // WriteJSON will error for us anyway
```

Comment:
On error can just do this:
```
log.Errorf(log.WebsocketMgr, "%s websocket connection: sending message: %+v %v\n", w.ExchangeName, data, err)
```

Reply:
But then we'd error twice, right?
And I didn't want divergent error paths for this with Verbose on/off.

Reply:
I was suggesting more of an `if else` just so whatever error state or data is? it's still being communicated in logs. But If you think this is too heavy handed then just resolve.